File: src/ceal.c
#include "../include/bcind_kernel.h"

bool ceal_enforce_policy(TelemetryFrame* frame, GovernanceState* state) {
    if (frame == NULL || state == NULL) {
        fprintf(stderr, "[CEAL] Error: Null pointer passed to policy enforcement.\n");
        return false;
    }

    if (!state->execution_permitted) {
        fprintf(stderr, "[CEAL] Enforcement Rejection: Governance state execution flag is disabled.\n");
        return false;
    }

    if (state->immutable_lock) {
        printf("[CEAL] Immutable lock active - strict policy enforcement engaged.\n");
    }

    return true;
}

File: src/ceal.rs
use crate::governance::GovernanceState;
use crate::law_envelope::TelemetryFrame;

pub fn ceal_enforce_policy(_frame: &TelemetryFrame, state: &GovernanceState) -> bool {
    if !state.execution_permitted {
        log::error!("[CEAL] Enforcement Rejection: Governance state execution flag is disabled.");
        return false;
    }

    if state.immutable_lock {
        log::info!("[CEAL] Immutable lock active - strict policy enforcement engaged.");
    }

    true
}
